lib/docker2aci.go

```diff
@@ -734,8 +743,9 @@ func mergeManifests(manifests []schema.ImageManifest) schema.ImageManifest {
 		manifest.Labels = append(manifest.Labels[:layerIndex], manifest.Labels[layerIndex+1:]...)
 	}
 
+	n := strings.LastIndex(manifest.Name.String(), "-")
 	// this can't fail because the old name is legal
-	nameWithoutLayerID, _ := types.NewACName(strings.Split(manifest.Name.String(), "-")[0])
```

Comment:
Every layer's name has the docker layerID at the end, this strips it from the squashed image name. It was wrong because the image name can have dashes.